lib/ruby_smb/client.rb

```diff
@@ -575,6 +575,9 @@ def recv_packet(encrypt: false)
       begin
         raw_response = dispatcher.recv_packet
       rescue RubySMB::Error::CommunicationError => e
+        # If raw_response is nil the connection is dead
+        raise EOFError if raw_response.nil?
```

Comment:
Generally third party libraries don't raise raw exceptions like this, normally they're encapsulated and namespaced by the gem in question

i.e. what happened here: 

https://github.com/rapid7/ruby_smb/blob/0611a3e5b745d1ae5e49cd3514ad7d3c30132811/lib/ruby_smb/dispatcher/socket.rb#L87-L88

I think this try/catch logic that you're trying to update is the real problem that needs to be re-evaluated